src/main/java/com/uber/cadence/worker/Worker.java

```diff
@@ -0,0 +1,236 @@
+                .setReportCompletionRetryOptions(options.getReportWorkflowCompletionRetryOptions())
+                .setReportFailureRetryOptions(options.getReportWorkflowFailureRetryOptions())
+                .setTaskExecutorThreadPoolSize(options.getMaxConcurrentWorklfowExecutionSize())
+                .build();
+    }
+
+    /**
+     * Register workflow implementation classes with a worker.
+     * A workflow implementation class must implement at least one interface with a method annotated with
+     * {@link com.uber.cadence.workflow.WorkflowMethod}. That method becomes a workflow type
+     * that this worker supports.
+     * <p>
+     * Implementations that share a worker must implement different interfaces as a workflow type
+     * is identified by the workflow interface, not by the implementation.
+     * </p><p>
+     * The reason for registration accepting workflow class, but not the workflow instance is
+     * that workflows are stateful and a new instance is created for each workflow execution.
+     * </p>
+     */
+    public void registerWorkflowImplementationTypes(Class<?>... workflowImplementationClasses) {
+        workflowWorker.setWorkflowImplementationTypes(workflowImplementationClasses);
+    }
+
+    /**
+     * Register activity implementation objects with a worker.
+     * As activities are reentrant and stateless only one instance per activity type
+     * is registered.
+     * <p>
+     * Implementations that share a worker must implement different interfaces as an activity type
+     * is identified by the activity interface, not by the implementation.
+     * <p>
+     * </p>
+     */
+    public void registerActivitiesImplementations(Object... activityImplementations) {
+        if (activityWorker == null) {
+            throw new IllegalStateException("disableActivityWorker is set in worker options");
+        }
+        checkStarted();
+        activityWorker.setActivitiesImplementation(activityImplementations);
+    }
+
+    private void checkStarted() {
+        if (started.get()) {
+            throw new IllegalStateException("already started");
+        }
+    }
+
+    public void start() {
+        if (!started.compareAndSet(false, true)) {
+            return;
+        }
+        if (workflowWorker != null) {
+            workflowWorker.start();
+        }
+        if (activityWorker != null) {
+            activityWorker.start();
+        }
+    }
+
+    /**
+     * Shutdown a worker, waiting for activities to complete execution up to the specified timeout.
+     */
+    public void shutdown(Duration timeout) {
+        try {
+            long time = System.currentTimeMillis();
+            if (activityWorker != null) {
+                activityWorker.shutdownAndAwaitTermination(timeout.toMillis(), TimeUnit.MILLISECONDS);
+            }
+            if (workflowWorker != null) {
+                long left = timeout.toMillis() - (System.currentTimeMillis() - time);
```

Comment:
left could be negative if activityWorker took too long to stop.

Reply:
shutdownAndAwaitTermination handles negative value gracefully.